runtime/js/40_testing.js

```diff
@@ -173,13 +173,16 @@
           preOp.opsCompletedAsync;
 
         if (dispatchedDiff > completedDiff) {
-          const [name, hint] = OP_DETAILS[key];
+          const [name, hint] = OP_DETAILS[key] || [key, null];
           const count = dispatchedDiff - completedDiff;
           let message = `${count} async operation${
             count === 1 ? "" : "s"
           } to ${name} ${
             count === 1 ? "was" : "were"
-          } started in this test, but never completed. This is often caused by not ${hint}.`;
+          } started in this test, but never completed.`;
```

Comment:
Personally I would go with an infernal nested template string:
```ts
`...but never completed.${hint ? ` This is often caused by not ${hint}.` : ""}
```
Buuut that's just personal preference and I admit that nested template strings are... uhm, not pretty to put it mildly.